x-pack/plugins/uptime/server/lib/adapters/monitors/elasticsearch_monitors_adapter.ts

```diff
@@ -97,32 +86,41 @@ export class ElasticsearchMonitorsAdapter implements UMMonitorsAdapter {
     };
 
     const result = await this.database.search(request, params);
-    const buckets = get(result, 'aggregations.timeseries.buckets', []);
+    const buckets = dropLatestBucket(get(result, 'aggregations.timeseries.buckets', []));
 
-    return buckets.map(
-      ({
-        key,
-        max_content,
-        duration: { avg, max, min },
-        max_write_request,
-        max_validate,
-        max_tcp_rtt,
-        max_response,
-        max_total,
-        status,
-        doc_count,
-      }: any) => ({
-        maxContent: formatChartValue(key, max_content),
-        maxWriteRequest: formatChartValue(key, max_write_request),
-        maxValidate: formatChartValue(key, max_validate),
-        maxTcpRtt: formatChartValue(key, max_tcp_rtt),
-        maxResponse: formatChartValue(key, max_response),
-        maxTotal: formatChartValue(key, max_total),
-        avgDuration: formatChartValue(key, { value: avg }),
-        maxDuration: formatChartValue(key, { value: max }),
-        minDuration: formatChartValue(key, { value: min }),
-        status: formatStatusBuckets(key, status.buckets, doc_count),
-      })
+    let statusMaxCount = 0;
+    let durationMaxValue = 0;
+    const mappedBuckets = buckets.map(bucket => {
+      const x = get(bucket, 'key');
+      const docCount = get(bucket, 'doc_count', 0);
+      statusMaxCount = Math.max(docCount, statusMaxCount);
```

Comment:
This needs comments for clarity, it's pretty confusing what's going on here. A comment saying "This can be used to calculate the maximum Y bounds for a chart" would be quite helpful.

Reply:
I added some comments in https://github.com/elastic/kibana/pull/30561/commits/3ccdd1dc85e8107a6d01a5808d84e7d2aac6635c, please elaborate if you still think it's unclear.